src/i_oalmusic.c

```diff
@@ -259,9 +259,7 @@ static void I_OAL_SetMusicVolume(int volume)
     if (!music_initialized)
         return;
 
-    SDL_LockMutex(music_lock);
     alSourcef(player.source, AL_GAIN, (ALfloat)volume / 15.0f);
-    SDL_UnlockMutex(music_lock);
```

Comment:
The game still pauses briefly during a song change.

> When the song changes we load and parse the whole MIDI file and midifile.c is not very fast. We can do this in the player thread.

As a test, I created a MIDI that plays a single piano note (48 byte file size) and placed it in the test wad and there was still a long pause (game freezes for a moment) to switch songs. I'm not sure if midifile.c is the reason.

> MS GS Wavetable init is 200+ ms on my system, VirtualMidiSynth - 4 ms. I guess another problem with the MS GS Wavetable.

MS GS Synth has problems but something else is going on (this happens with every backend). I cloned the master branch and reverted these two commits and everything is back to normal:
- https://github.com/fabiangreffrath/woof/commit/c744db3a86361a4f1d351fc28465ef53e07a83c9
- https://github.com/fabiangreffrath/woof/commit/fc9731ffa02ea376a9c28cff47083ca03a438902

I haven't looked closer to see why, but that should narrow it down. The other commits since the 12.0 release don't appear to be related.

Reply:
I have removed the mutex here: https://github.com/fabiangreffrath/woof/pull/1430/commits/ecc2b2b377b99c30bd237ed4326c3936be35f324 It's strange that it works on my system.

Reply:
I will try to test again when I'm free. Sorry for the delay.

Reply:
> > MS GS Wavetable init is 200+ ms on my system, VirtualMidiSynth - 4 ms. I guess another problem with the MS GS Wavetable.
> 
> MS GS Synth has problems but something else is going on (this happens with every backend).

I'm talking about changing MIDI devices. Is MS GS Wavetable fast on your system? For me it's about 100 times slower than other WINMM MIDI devices.

Reply:
> I'm talking about changing MIDI devices. Is MS GS Wavetable fast on your system? For me it's about 100 times slower than other WINMM MIDI devices.

When changing MIDI devices from the menu, MS GS Synth is indeed significantly slower to initialize than other backends on my system. It doesn't seem 100 times slower, but I haven't timed it.

Reply:
And yes, MS GS Synth is slower to initialize than other WinMM devices as well (VMS and loopMIDI+FSMP are faster).

Reply:
> I have removed the mutex here: [ecc2b2b](https://github.com/fabiangreffrath/woof/commit/ecc2b2b377b99c30bd237ed4326c3936be35f324) It's strange that it works on my system.

With this change, there is still a brief pause (game freezes for a moment) but it's not as severe as before. Some backends seem worse than others. For my system, it's more pronounced with FluidSynth, but they are all affected. Are you able to notice it if you strafe smoothly across the sectors?